src/cart/crt_swim.c

```diff
@@ -1005,6 +1005,27 @@ static void crt_swim_update_last_unpack_hlc(struct crt_swim_membs *csm)
 	D_RWLOCK_UNLOCK(&crt_gdata.cg_rwlock);
 }
 
+static void
+crt_metrics_sample_delay(crt_context_t crt_ctx, uint64_t delay, bool glitch)
+{
+	struct crt_context *ctx;
+
+	if (unlikely(crt_ctx == CRT_CONTEXT_NULL)) {
+		D_ERROR("invalid parameter (NULL crt_ctx).\n");
+		return;
+	}
+
+	ctx = crt_ctx;
+
+	if (delay == 0)
+		d_tm_zero_gauge(ctx->cc_swim_delay);
+	else
+		d_tm_set_gauge(ctx->cc_swim_delay, delay);
```

Comment:
```suggestion
	d_tm_set_gauge(ctx->cc_swim_delay, delay);
```
Is there anything special about setting a gauge to 0, vs. any other value?

Reply:
If the gauge has stats, then setting it to zero repeatedly will recompute the stats (see [d_tm_compute_stats()](https://github.com/daos-stack/daos/blob/master/src/gurt/telemetry.c#L1665)), and my thought was that it's not that useful to do so for this particular metric (it will usually be zero outside of transient network events). So the new function sets the value to zero and recomputes the stats only if the value is not already zero.

Reply:
In my testing, I have observed that this value is normally zero. By normally, I mean 99.999% of the time that a system is running, assuming that everything is working correctly. IMO, it's not particularly useful to measure this state of affairs. In other words, I (and I don't think anyone else) cares to know how many zero values were sampled, so there's no point in doing the stats update which increments the `sample_size` field and all of the other work.

In the case where things have gone wrong, however, the values ARE useful to see, and the min/max/mean stats are helpful to see how bad things were during the event that caused the non-zero SWIM delay readings. If we continue to increment the `sample_size` field with subsequent zero readings, then the mean value very quickly decays away to almost nothing, and therefore it's hard to see it on a graph.

All that having been said, I can see the argument that mean decaying to nothing is a reflection of reality, and that it's probably a better idea to use the other values to compute what I want to see on the graph rather than modifying things here. I'll revert this part of the patch and see if I can get what I want in a different way. I'd really like to get this landed in some form so that I can close the tracking bug.